Enhancer/PayloadEnhancer.php

```diff
@@ -24,9 +24,15 @@ class PayloadEnhancer implements EnhancerInterface
     /**
      * {@inheritDoc}
      */
-    public function enhance(Context $context, Resource $resource)
+    public function enhance(array $data, Context $context, Resource $resource)
     {
         $visitor = $context->getVisitor();
-        $visitor->addData('payload', $context->accept($resource->getPayload()));
+        $payload = $resource->getPayload();
+
+        $context->startVisiting($payload);
+        $data['payload'] = $context->accept($payload);
```

Comment:
Cannot get this to work at all.. I think the solution is to basically inject the serializer and use it directly from the "outside".